code/scripting/api/objs/model.cpp

```diff
@@ -473,6 +485,38 @@ ADE_FUNC(isValid, l_Submodel, nullptr, "True if valid, false or nil if not", "bo
 	return ade_set_args(L, "b", smh->isValid());
 }
 
+ADE_VIRTVAR(NoCollide, l_Submodel, nullptr, "Whether the submodel and its children ignore collisions", "boolean", "The flag, or error-false if invalid")
+{
+	submodel_h* smh = nullptr;
+
+	if (!ade_get_args(L, "o", l_Submodel.GetPtr(&smh)))
+		return ade_set_error(L, "b", false);
+
+	if (!smh->isValid())
+		return ade_set_error(L, "b", false);
+
+	if (ADE_SETTING_VAR)
+		smh->GetSubmodel()->flags.set(Model::Submodel_flags::No_collisions);
```

Comment:
I am really not a fan of allowing scripts to set values in static data like the models themselves. Is there a reason this couldn't be a flag/function on a `submodel_instance` instead? This would also make it more useful in general, since you could toggle specific ship submodels on and off, rather than all ships using that model.

Reply:
Whoops.  Now it handles setting the flag.

On the one hand, I ended up using a different strategy in the script I was writing, so toggling the flag is no longer needed for my particular use case.  On the other hand, this might still be useful for another script.

This can't be a flag on `submodel_instance` because of the way the collision code works.  Not every model has a model instance (and submodel instances), although ships do.  More importantly, if I understand the collision code correctly, submodel instances are only processed if the submodel is actually moving.

The reason for allowing the flag to be set is to give scripts more control over collision detection.  The idea is to let the script toggle the flag, do a collision check, and then put the flag back the way it was.  For Event Horizon, this would allow flying ships to collide with the ground, while preventing land-based ships from colliding with it.  (I ended up using the +Override mechanism instead.)

Reply:
It is reassuring to know there are 'safe' uses, and that *you* intend to use it safely, but the flag is not required to be used that way, and so remains something of a footgun. A footgun is not called that way because *there merely exist* safe uses of it, but conversely because *there merely exist* unsafe ways to use it.

Calling it a footgun or unsafe might be slightly hyperbolic, but I'm still not entirely convinced the utility outweighs the hassle of getting it to work right in what the user might otherwise see as a simple use case, given how much of FSO internals with regards to models it implicitly exposes. Escpecially considering, again, it would apply to *all* uses of the model, *even entirely different ship classes*.

I don't want to embroil you in a fight over this forever, so it might be worth getting other reviewers to chime in, If others disagree, then it is what it is.

Reply:
Well Lafiel already said separately that he broadly agrees with your comments.  I'll remove the ability to set the flag.  It's no longer necessary for my use case, and I would hope that other use cases could similarly find an alternate way to accomplish what they need.